src/Symfony/Component/Messenger/Command/CountMessagesCommand.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Command;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+
+/**
+ * @author Kévin Thérage <therage.kevin@gmail.com>
+ */
+#[AsCommand(name: 'messenger:count', description: 'Show the message count for one or more transports')]
+class CountMessagesCommand extends Command
+{
+    private ContainerInterface $transportLocator;
+    private array $transportNames;
+
+    public function __construct(ContainerInterface $transportLocator, array $transportNames = [])
+    {
+        $this->transportLocator = $transportLocator;
+        $this->transportNames = $transportNames;
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->addArgument('transportNames', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, 'Name or list of transports\' names to get the count')
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> command counts the messages for all the transports:
+
+    <info>php %command.full_name%</info>
+
+Or a specific transports only:
+
+    <info>php %command.full_name% <transportNames></info>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+
+        $transportNames = $this->transportNames;
+        if ([] !== $input->getArgument('transportNames')) {
+            $transportNames = $input->getArgument('transportNames');
+        }
+
+        $outputTable = [];
+        foreach ($transportNames as $transportName) {
+            if (!$this->transportLocator->has($transportName)) {
+                $io->warning(sprintf('The "%s" transport does not exist.', $transportName));
+
+                continue;
+            }
+            $transport = $this->transportLocator->get($transportName);
+            if (!$transport instanceof MessageCountAwareInterface) {
+                $io->warning(sprintf('Unable to get message count from transport "%s".', $transportName));
```

Comment:
A simple note might be enough semantically, there is nothing abnormal here, but that's a good idea to indicate to the user these transports do not support fetching the number of messages in it.

A single line with the names of "uncountable" transports at the end of the command output might be better, perhaps.

Reply:
I've fixed this.